wts/api.py

```diff
@@ -78,19 +78,32 @@ def load_settings(app):
             state_prefix = ""
 
         for idp, idp_conf in conf.get("login_options", {}).items():
-            authorization_url = fence_base_url + "oauth2/authorize"
-            authorization_url = add_params_to_uri(
-                authorization_url, idp_conf.get("params", {})
-            )
+            scope = "openid data user"
+            idp_params = idp_conf.get("params", {})
+            idp_client = idp_params.get("idp", "")
+            if idp_client == "keycloak":
```

Comment:
I don't like the idea of hard-coding this name here. This should at least a global constant, or a static class constant (see below idea about factory pattern), or configuration or something. 

I'm not sure how well this pattern scales to when we add more IdPs. I don't really want to have an if/elif/elif with big blocks of code here for every IdP. We need maybe a simple factory class pattern or at a minimum: break these code blocks out to functions so it's just if: `get_cfg_from_keycloak` else: `get_cfg_from_default` or something like that

Reply:
i think you could remove the `idp = keyclock` field from the config and do this:
```
if "auth_url" in idp_params and "token_url" in idp_params:
    < new logic here >
else:  # default to Gen3 fence integration
    < existing logic here >
app.config["OIDC"][idp] = { ... }
```